plugins/configuration.rb

```diff
@@ -0,0 +1,92 @@
+module Jekyll
+  class ConfigurationBlock < Liquid::Block
+    TYPE_LINKS = {
+      'action'       => '/docs/automation/action/',
+      'device_class' => '/components/binary_sensor/',
+      'template'     => '/docs/configuration/templating/',
+    }
+
+    REQUIRED_VALUES = {
+      true        => 'Required',
+      false       => 'Optional',
+      'required'  => 'Required',
+      'optional'  => 'Optional',
+      'inclusive' => 'Inclusive',
+      'exclusive' => 'Exclusive'
+    }
+
+    def initialize(tag_name, text, tokens)
+      super
+    end
+
+    def slug(key)
+      key.downcase.strip.gsub(' ', '-').gsub(/[^\w-_]/, '')
+    end
+
+    def type_class(type)
+      ((type.is_a? Array) ? type.join(' ') : type).downcase
+    end
+
+    def type_link(type)
+      if type.include? ','
+        type = type.split(',')
+      end
+
+      if type.is_a? Array
+        return (type.map { |t| type_link(t) }).join(' | ')
+      end
+
+      type = type.strip
+      (TYPE_LINKS.include? type.downcase) ? "[%s](%s)" % [type, TYPE_LINKS[type.downcase]] : type
+    end
+
+    def required_value(value)
+      if !!value == value
+        return REQUIRED_VALUES[value]
+      end
+
+      value = value.strip.downcase
+      (REQUIRED_VALUES.include? value) ? REQUIRED_VALUES[value] : REQUIRED_VALUES[false]
+    end
+
+    def render_config_vars(vars, nested = false)
+      result = Array.new
+      result << "<dl class='#{nested ? "nested" : ""}'>"
+
+      result << vars.map do |key, attr|
+        markup = Array.new
+        markup << "<dt><a class='title-link' name='#{slug(key)}' href='\##{slug(key)}'></a> #{key}</dt>"
```

Comment:
Ruby supports multi-line strings.

```ruby
markup = <<TPL
Here your fancy
And multi-line string including interpolation: #{slug(key)}
TPL
```

Reply:
The reason I broke it up like this is because of all of the `if` statements. I could break it into an array, and then build from there, but I do not know how to do the `if` statements inside of a heredoc.

Reply:
Oh I didn't even see the if statements.